src/executor/sort_merge_join.rs

```diff
@@ -0,0 +1,217 @@
+                                left_current_chunk = chunk?;
+                            } else {
+                                break;
+                            }
+                        }
+                        Ordering::Greater => {
+                            if let Some(chunk) = right_chunks.next().await {
+                                right_current_chunk = chunk?;
+                            } else {
+                                break;
+                            }
+                        }
+                        Ordering::Less => {
+                            if let Some(chunk) = left_chunks.next().await {
+                                left_current_chunk = chunk?;
+                            } else {
+                                break;
+                            }
+                        }
+                    }
+                }
+                // if rows line < PROCESSING_WINDOW_SIZE ,take rest rows out of builder
+                if let Some(chunk) = { builder }.take() {
+                    yield chunk;
+                }
+            }
+            _ => yield Err(ExecutorError::NotNullable)?,
+        }
+    }
+}
+// compare two rows by join key
+pub fn compare_row(
+    left_row: &Row,
+    right_row: &Row,
+    left_column_index: usize,
+    right_column_index: usize,
+) -> Ordering {
+    let left_data_value = &left_row[left_column_index];
+    let right_data_value = &right_row[right_column_index];
+    left_data_value.partial_cmp(right_data_value).unwrap()
+}
+
+// convert chunk stream to row stream
+#[try_stream(boxed, ok = Row, error = ExecutorError)]
+async fn row_stream(stream: BoxedExecutor) {
+    #[for_await]
+    for chunk in stream {
+        for row in chunk?.rows() {
+            yield row.values().collect();
+        }
+    }
+}
+
+// chunk same join key rows together
+#[try_stream(boxed, ok = Vec<Row>, error = ExecutorError)]
+async fn same_key_chunks(
+    row_stream: BoxStream<'static, Result<Row, ExecutorError>>,
+    column_index: usize,
+) {
+    let mut current_row = None;
+    let mut chunk = Vec::new();
+    #[for_await]
+    for row in row_stream {
+        match row {
+            Ok(row) => {
+                if current_row.is_none() {
+                    current_row = Some(row);
+                    chunk.push(current_row.clone().unwrap());
+                    continue;
+                }
+                if current_row.as_ref().unwrap().get(column_index) == row.get(column_index) {
+                    chunk.push(row);
+                } else {
+                    yield chunk;
+                    chunk = Vec::new();
+                    current_row = Some(row);
+                    chunk.push(current_row.clone().unwrap());
+                }
+            }
+            Err(e) => {
+                yield Err(e)?;
+            }
+        }
+    }
+    if !chunk.is_empty() {
+        yield chunk;
+    } else {
+        yield Err(ExecutorError::NotNullable)?;
```

Comment:
No need to error, just return?

Reply:
I change return chunk directly `    yield chunk; ` Empty chunk is acceptable